### ISSUE TITLE
[bug] conan conf install issue when executing across different users.

### ISSUE THREAD

**Issue by fulara** [conan-io/conan#6556]
It seems that conan config install requires to be an owner of settings.yml before invoking conan config - even though the settings.yml has rwx accesses.
gives a cryptic error if its not an owner.

What am i doing?
In our complex world we have a two separate users that can deploy conan onto boxes, 

as a part of our deploy procedure I am doing conan config install
it reports an error:
[Errno 1] Operation not permitted: '.../deploy/.conan/settings.yml'

My guess is that it does chmod on settings.yml rather than a simple modification or simple 'cp' over it.

It would be nice if error would be more descriptive.

Anyway, it does not seem like a chmoding this file should be part of the procedure if the file existed historically?
Or check if permissions are good enough and then skip this step?

As a workaround before invoking conan install we are now removing settings.yml file.

### Environment Details (include every applicable attribute
  * Operating System+version: redhat6

**Comment by memsharded** [comment]
The operation that could raise this error is:

```python
 if f == "settings.yml":
                output.info("Installing settings.yml")
                settings_path = cache.settings_path
                shutil.copy(os.path.join(root, f), settings_path)
```

So, basically a python ``shutil.copy``, nothing special in Conan. I am not comfortable with any automatic procedure that changes the permissions or the owner of that file, in order to be able to overwrite its contents.

Skipping updating the settings.yml that you are trying to ``conan config install`` doesn't seem right either.

Yes, we could capture exceptions and be more clear in the error message, but do you think that Conan could do something more?

**Comment by fulara** [comment]
Hello @memsharded ,

I've just done a quick test of this, 
And basically - there would be no issue for me if you'd use `shutil.copyfile` rather than `shutil.copy`

See below:
Considering a directory:
```
[fulara@localhost testbench]$ ls -lah
-rw-rw-rw-.  1 test-user test-user    6 Feb 19 19:43 file
-rw-rw-r--.  1 fulara    fulara       6 Feb 19 19:40 file2
```
as fulara i will invoke:
```
>>> import shutil
>>> shutil.copyfile("file2", "file")
'file'
``` 
This works fine, however when you try this with `copy` you'd get:
```
>>> import shutil
>>> shutil.copy("file2", "file")
Traceback (most recent call last):
...
PermissionError: [Errno 1] Operation not permitted: 'file'
```

Looking at the docs:
```
shutil.copy(src, dst)

    Copy the file src to the file or directory dst. If dst is a directory, a file with the same basename as src is created (or overwritten) in the directory specified. Permission bits are copied. src and dst are path names given as strings.
```


I reckon that if shutil.copyfile works then thats good enough, hence I'd go with shutil.copyfile.
Additionally it seems that `shutil.copyfile` behaves the same way as `cp` does in this case.


re:
 I am not comfortable with any automatic procedure that changes the permissions or the owner of that file, in order to be able to overwrite its contents.
Sorry - this is because I wasnt clear, permissions are all good for me - I dont want them changed, its just that 'cp' worked for me but conan conf install didnt so, that was super confusing.

**Comment by memsharded** [comment]
I have submitted PR https://github.com/conan-io/conan/pull/6601 to check.
My feeling is that this is not really a bug. You are trying to overwrite some file with another user that doesn't have the permissions to change the file mode (but it can write it). Conan tries to keep the installed file with its permission/mode, because it seems that such behavior is what users would expect. The proposed PR could take advantage that removal is actually allowed, but please check for your use case.

In any case, I am curious of why would you be using ``conan config install`` from 2 different users over the same cache. Conan has been always designed at the user level, it never considered sharing the same cache for 2 different users. Could you please explain a bit more?

**Comment by fulara** [comment]
Sure @memsharded 

I originally wrote 'because we live in a complex word...' and indeed thats pretty much the reasonw why :).

This cache is used on the target box, where applications are running, and some of these boxes we are sharing across different teams.  
These teams have different deploy users, they all belong to a group 'deploy' though.
It has been agreed that we will share cache on these machines, mainly due to disk space restrictions (sigh.) 

What we are doing right now, is something like this.. 
So any of these teams are free to deploy conan onto these boxes themselves, however we share a common setup script.  this script does invokes:
```
setfacl -Rdm g:deploy:rwx ....
setfacl -Rm g:deploy:rwx ....
```
And thanks to that all the users of group deploy are able to meddle within the conan cache. 

As to your argumentation, I am slightly confused with that, I think its reasonable for me to expect that `conan conf install` will succeed if particular directory is writable. 
`cp` command does exactly that - it writes contents, but it does not overwrite user when you copy onto a file that has permissions, but is not yours.

**Comment by jgsogo** [comment]
Before reviewing #6601 we need to agree on the expected behavior. I assume we all agree that the new file has to be copied, but there are two approaches:
 1. Remove and copy
 2. Copy new content preserving existing user and permissions

IMO, (2) would be preferred, and if `shutil.copyfile` works, why not implementing it in those terms instead of #6601?

**Comment by memsharded** [comment]
My rationale is that it would be more intuitive. If the file that is on disk has been retrieved from a given repo or zip file, I would expect the file permissions to match those of the origin. Having different file mode, permissions, or whatever, simply because there was other file in the past with that mode, seems more difficult to understand and debug. The file contents and the file mode should come from the same place, IMHO.

**Comment by fulara** [comment]
Slightly extreme, but this is indeed what we are doing right now.

We are removing this single file settings.yml and then invoking `conan install`.

**Comment by jgsogo** [comment]
Ok, probably the way to do it is not very important, but it has to work. It is probably easier to implement too, so:

-> Remove and copy <-

**Comment by memsharded** [comment]
Merged #6601 will be released in 1.24
